Fixes

clusterfack opened the pull request:
Painstakingly goes through the 'in world' commit again to use the proper lists for all objects that are autoremoved from the machines list.

Also removes an extremely laggy proc called get_apc() which called locate throughout all related areas until it found an apc, instead apcs are linked to the areas they are in as only one is allowed per area in any case.

clusterfack commented:
Question: do lighting subareas get destroyed? Do areas get a destroy proc?

ComicIronic commented on code/modules/admin/topic.dm:
```diff
@@ -2506,7 +2506,7 @@
 				feedback_inc("admin_secrets_fun_used",1)
 				feedback_add_details("admin_secrets_fun_used","WO")
-				for(var/obj/machinery/light/L in machines)
+				for(var/obj/machinery/light/L in world)
```
What

clusterfack replied:
These aren't in machines

d3athrow replied:
Then put them in their own list nerd

d3athrow commented:
>Question: do lighting subareas get destroyed? Do areas get a destroy proc? These mysteries haunt me when I sleep.

Don't know

DeityLink commented:
It is a mystery to everyone

clusterfack commented:
Addressed, also added a destroy proc for lighting subareas to null the references on them. Shouldn't matter either way if they never get deleted and will be helpful if they do.

PJB3005 commented:
I unlaggied get_apc() in the lighting PR.

clusterfack commented:
That may be awhile before being fully finished and tested however so I just threw it in here

ComicIronic commented:
IN WARUDO

RemieRichards commented:
Either Github is displaying this weird or you're not actually using the type iterated in the loops? (A)
Am I just going mad here?

clusterfack commented:
Are you commenting about a specific part or...?

RemieRichards commented:
well, all of ```set_apc()``` ```remove_apc()``` ```and area/Destroy()```

clusterfack commented:
I'm not sure what you mean, it is updating the related lighting subareas to also have the reference to the APC, and removing it on APC or lighting subarea destruction.

clusterfack commented:
Ohhh I see what you mean

RemieRichards commented:
Right, take area/Destroy()

```
/area/Destroy()
	..()
	for(var/area/A in src.related)
		related -= src
	areaapc = null
```
A is unused there, completely.

```
/area/proc/set_apc(var/obj/machinery/power/apc/apctoset)
	for(var/area/A in src.related)
		areaapc = apctoset

```
A also unused, this looks like the entire proc could just be ```areaapc = apctoset```
```
/area/proc/remove_apc(var/obj/machinery/power/apc/apctoremove)
	for(var/area/A in src.related)
		if(areaapc == apctoremove)
			areaapc = null
```
A, again unused, also looks like a simple ```if(areaapc == apctoremove) areaapc = null``` would do

So again, is this some weird design or github being a shit?

clusterfack commented:
Yeah I just didn't add A. like I meant

RemieRichards commented:
Figured.
